### ISSUE TITLE
Try to make moist thermo work for unphysical values

### ISSUE THREAD

**Issue by charleskawczynski** [climate-machine/CLIMA#720]
In the case of RK schemes, it not unreasonable for intermediate quantities to have unphysical values (e.g., velocities that do not satisfy conservation, negative `q_tot`). This seems to be a good reason to allow for moist-thermo to not break for slightly negative `q_tot`, for example. Furthermore, this is low hanging fruit since it can be analyzed offline whereas positivity preserving schemes are involved.

For more info, see #719

**Comment by tapios** [comment]
Isn't the fix you had, setting `q_tot = max(0, q_tot)` for thermodynamics only, taking care of this?

**Comment by charleskawczynski** [comment]
Yes, good point. I guess my thought was that we could apply this to all of the inputs, but maybe that doesn't make sense.

**Comment by charleskawczynski** [comment]
Partially addressed in #1547
